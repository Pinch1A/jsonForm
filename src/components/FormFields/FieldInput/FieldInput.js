import React from 'react'
import styled from 'styled-components'
import { useInput } from '../../../hooks/useInput'

const InputField = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 300px;
  padding: 20px;
`

const InputLabel = styled.div`
  display: flex;
`

const FieldInput = ({ name, placeholder, label, required, type, conditions }) => {
  const keyField = name || label
  const { value, bind } = useInput({ [keyField]: '' })

  return (
    <InputField>
      <InputLabel>
        {label} {required && <span>*</span>}
      </InputLabel>
      <input
        type={type}
        {...bind}
        name={name}
        placeholder={placeholder}
        value={value[name]}
        required={required}
      />
    </InputField>
  )
}

export default FieldInput
