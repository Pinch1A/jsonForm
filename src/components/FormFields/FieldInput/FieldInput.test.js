import FieldInput from './FieldInput'
import React from 'react'
import { shallow, mount } from 'enzyme'

const props = {
  name: 'Test Field',
  placeholder: 'Placeholder',
  label: 'Test Label',
  required: true,
  type: 'text',
  conditions: [],
}

describe('<FieldInput>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<FieldInput {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('FieldInput - functionalities: ', () => {
    it('renders the label', () => {
      const wrapper = mount(<FieldInput {...props} />)
      expect(wrapper.text()).toContain(props.label)
    })
    it('renders the input and updates it', () => {
      const wrapper = mount(<FieldInput {...props} />)
      const inputField = wrapper.find('input')
      expect(inputField.render().attr('type')).toEqual(props.type)
      expect(inputField.render().attr('value')).toEqual('')
      inputField.simulate('change', { target: { value: 'Hello' } })
      wrapper.update()
      expect(inputField.render().attr('value')).toEqual('Hello')
    })
  })
})
