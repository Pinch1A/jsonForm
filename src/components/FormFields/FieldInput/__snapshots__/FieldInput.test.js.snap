// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldInput> FieldInput - functionalities:  renders without crashing 1`] = `ShallowWrapper {}`;

exports[`<FieldInput> renders without crashing 1`] = `ShallowWrapper {}`;
